src/mscorlib/corefx/System/Security/SecureString.Windows.cs

```diff
@@ -143,6 +144,49 @@ private void SetAtCore(int index, char c)
             }
         }
 
+        internal unsafe IntPtr MarshalToBSTR()
+        {
+            int length = _decryptedLength;
+            IntPtr ptr = IntPtr.Zero;
+            IntPtr result = IntPtr.Zero;
+            byte* bufferPtr = null;
+
+            UnprotectMemory();
+            try
+            {
+                _buffer.AcquirePointer(ref bufferPtr);
+                int resultByteLength = (length + 1) * sizeof(char);
+
+                ptr = Win32Native.SysAllocStringLen(null, length);
+                if (ptr == IntPtr.Zero) {
+                    throw new OutOfMemoryException();
+                }
+
+                Buffer.MemoryCopy(bufferPtr, (byte*)ptr, resultByteLength, length * sizeof(char));
+
+                *(length + (char*)ptr) = '\0';
```

Comment:
BSTR *is* null terminated. Check https://msdn.microsoft.com/en-us/library/windows/desktop/ms221069(v=vs.85).aspx.